tests/test_Source/test_source_variability.py

```diff
@@ -26,6 +27,34 @@ def test_variability_at_t_sinusoidal(self):
         expected_result = np.array([-0.43030122, -0.97536797, -0.14773276])
         npt.assert_almost_equal(result, expected_result, decimal=5)
 
+    def test_interpolation_for_sinusoidal(self):
+        kwargs_model00 = {"amp": 1.0, "freq": 0.5}
```

Comment:
can you also describe what you are testing here? This might help the reader and assess the completeness of the test @Henry-Best-01

Reply:
I added details to my tests (all moved into test_param_until)